src/frontend/staff-portal/src/app/components/ticket-info/ticket-info.component.ts

```diff
@@ -1,147 +1,575 @@
+    // with violation Ticket form fields only that were changed
+    let putDispute = this.lastUpdatedDispute;
+
+    putDispute.violationTicket.ticketNumber = this.form.get('violationTicket').get('ticketNumber').value;
+    putDispute.violationTicket.surname = this.form.get('violationTicket').get('surname').value;
+    putDispute.violationTicket.givenNames = this.form.get('violationTicket').get('givenNames').value;
+    putDispute.violationTicket.driversLicenceNumber = this.form.get('violationTicket').get('driversLicenceNumber').value;
+    putDispute.violationTicket.driversLicenceProvince = this.form.get('violationTicket').get('driversLicenceProvince').value;
+
+    // provincial court hearing is returned by the GET at the level of notice of dispute but is to be saved with
+    // violation Ticket fields not notice of dispute fields
+    putDispute.provincialCourtHearingLocation = this.form.get('violationTicket').get('provincialCourtHearingLocation').value;
+
+    // reconstruct issued date as string from violation date and violation time format yyyy-mm-ddThh:mm
+    putDispute.violationTicket.issuedDate =
+      this.form.get('violationTicket').get('violationDate').value +
+      "T" +
+      this.form.get('violationTicket').get('violationTime').value.substring(0, 2)
+      + ":" +
+      this.form.get('violationTicket').get('violationTime').value.substring(2, 4);
+
+    // Loop through violation Ticket counts and set fields
+    putDispute.violationTicket.violationTicketCounts.forEach(violationTicketCount => {
+      if (violationTicketCount.count == 1) {
+        violationTicketCount = this.constructViolationTicketCount(this.form.get('violationTicket').get('violationTicketCount1').get('fullDescription').value);
+        violationTicketCount.ticketedAmount = this.form.get('violationTicket').get('violationTicketCount1').get('ticketedAmount').value;
+      } else if (violationTicketCount.count == 2) {
+        violationTicketCount = this.constructViolationTicketCount(this.form.get('violationTicket').get('violationTicketCount2').get('fullDescription').value);
+        violationTicketCount.ticketedAmount = this.form.get('violationTicket').get('violationTicketCount2').get('ticketedAmount').value;
+      } else if (violationTicketCount.count == 3) {
+        violationTicketCount = this.constructViolationTicketCount(this.form.get('violationTicket').get('violationTicketCount3').get('fullDescription').value);
+        violationTicketCount.ticketedAmount = this.form.get('violationTicket').get('violationTicketCount3').get('ticketedAmount').value;
+      }
+    });
+
+    this.putDispute(putDispute);
+  }
+
+  public onSubmitNoticeOfDispute(): void {
+    // We are only sending the notice of dispute fields so update a local copy of lastUpdatedDispute
+    // with notice of dispute form fields only that were changed
+    let putDispute = this.lastUpdatedDispute;
```

Comment:
same as contact info

Reply:
because your form only has the fields that you want and not disabled. You should be able to do something like
let putDispute = { ...this. lastUpdatedDispute, ...this.form.value }

It can replace around 10 lines of code